src/extension/background/vault-manager.ts

```diff
@@ -44,27 +48,38 @@ let vault: Vault = {
   ...defaultVault,
   encryptedSecretKey: localStorage.getItem(encryptedKeyIdentifier) || undefined,
   hasSetPassword: getHasSetPassword(),
+  salt: localStorage.getItem(saltIdentifier) || undefined,
 };
 
-export const getVault = () => {
-  return vault;
-};
+export const getVault = () => vault;
 
-export const vaultMessageHandler = async (message: MessageFromApp) => {
-  vault = await vaultReducer(message);
-  if (vault.encryptedSecretKey) {
-    localStorage.setItem(encryptedKeyIdentifier, vault.encryptedSecretKey);
+function persistOptional(storageKey: string, value?: string) {
+  if (value) {
+    localStorage.setItem(storageKey, value);
   } else {
-    localStorage.removeItem(encryptedKeyIdentifier);
+    localStorage.removeItem(storageKey);
   }
+}
+
+export async function vaultMessageHandler(message: MessageFromApp) {
+  vault = await vaultReducer(message);
+  persistOptional(encryptedKeyIdentifier, vault.encryptedSecretKey);
```

Comment:
Maybe local storage could be wrapped as promise that fails? This method is a side effect.

Reply:
What exactly are we trying to catch for? I'm not sure why `localStorage.setItem` would fail

Reply:
You're right, I guess set/remove methods are fine. Maybe it's okay, as I'm thinking of cases where you'd `JSON.parse` an unknown value